core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -549,6 +525,89 @@ public void addIndex(CharSequence columnName, int indexValueBlockSize) {
         LOG.info().$("ADDED index to '").utf8(columnName).$('[').$(ColumnType.nameOf(existingType)).$("]' to ").$(path).$();
     }
 
+    public int attachPartition(long timestamp) {
+        if (partitionBy == PartitionBy.NONE) {
+            return TABLE_NOT_PARTITIONED;
```

Comment:
non-happy paths are not tested (not just this one, almost none)

Reply:
ok, more unhappy tests added